tests/test_load.py

```diff
@@ -247,6 +247,24 @@ def test_load_dataset_streaming_gz_json(jsonl_gz_path):
     assert ds_item == {"col_1": "0", "col_2": 0, "col_3": 0.0}
 
 
+@require_streaming
+@pytest.mark.parametrize(
+    "path", ["sample.jsonl", "sample.jsonl.gz", "sample.tar", "sample.jsonl.xz", "sample.zip", "sample.jsonl.zst"]
+)
+def test_load_dataset_streaming_compressed_files(path):
+    repo_id = "albertvillanova/datasets-tests-compression"
+    data_files = f"https://huggingface.co/datasets/{repo_id}/resolve/main/{path}"
```

Comment:
This is a really nice feature !

I think the glob logic has to be moved in a data files resolution module, as done in https://github.com/huggingface/datasets/pull/2662

The current implementation may not be robust enough to work with path manipulations by users in compressed files

Reply:
I have not touched the glob logic in this PR though... 🤔